### PR TITLE
Handle filter type

### PR DESCRIPTION
**Pull request by curzio-della-santa**

### CODE REVIEW

File: src/Detail/Commanding/Command/Listing/Filter.php [hunk 1]
```diff
@@ -116,4 +118,37 @@ public function setType($type)
     {
         $this->type = $type;
```
Comment (ivan-wolf) [inline src/Detail/Commanding/Command/Listing/Filter.php]: What happens to the value when the type is changed?

Comment (curzio-della-santa) [inline src/Detail/Commanding/Command/Listing/Filter.php]: we have 2 possibilities here:
- work with a raw_value
- or change visibility of setType to private

Comment (ivan-wolf) [inline src/Detail/Commanding/Command/Listing/Filter.php]: I recommend introducing type as second (optional) parameter to setValue().

Comment (curzio-della-santa) [inline src/Detail/Commanding/Command/Listing/Filter.php]: Good point, implemented.

File: src/Detail/Commanding/Command/Listing/Filter.php [hunk 2]
```diff
@@ -95,10 +105,16 @@ public function getValue()
 
     /**
      * @param mixed $value
+     * @param string $type
```
Comment (ivan-wolf) [inline src/Detail/Commanding/Command/Listing/Filter.php]: |null

File: src/Detail/Commanding/Command/Listing/Filter.php [hunk 3]
```diff
@@ -31,7 +31,21 @@ class Filter
     /**
      * @var string
```
Comment (ivan-wolf) [inline src/Detail/Commanding/Command/Listing/Filter.php]: |null

File: src/Detail/Commanding/Command/Listing/Filter.php [hunk 4]
```diff
@@ -31,7 +31,21 @@ class Filter
...
+            'boolean' => array('bool', 'boolean'),
+            'integer' => array('int', 'digit', 'integer'),
+            'float' => array('float', 'decimal', 'double', 'real'),
+            'string' => array('str', 'string', 'uuid'),
+            'array' => array('array', 'hash'),
+        );
+    }
 
     /**
      * @param string $property
```
Comment (ivan-wolf) [inline src/Detail/Commanding/Command/Listing/Filter.php:53]: |null for operator  and type

File: src/Detail/Commanding/Command/Listing/Filter.php [hunk 5]
```diff
@@ -95,10 +105,16 @@ public function getValue()
...
-    public function setValue($value)
+    public function setValue($value, $type = null)
     {
-        $this->value = $value;
+        if ($type !== null) {
+            $this->setType($type);
+        }
+
+        // Has to be done after the type has been set (if any)
+        $this->value = $this->castToType($value);
```
Comment (ivan-wolf) [inline src/Detail/Commanding/Command/Listing/Filter.php]: Can't cast when there's no type.

File: src/Detail/Commanding/Command/Listing/Filter.php [hunk 6]
```diff
@@ -110,10 +126,26 @@ public function getType()
...
+        return $value;
+    }
+
+    /**
      * @param string $type
      */
-    public function setType($type)
+    private function setType($type)
     {
+        /** @todo Check that is one of the supported types first? */
```
Comment (ivan-wolf) [inline src/Detail/Commanding/Command/Listing/Filter.php]: Sure, check it.

File: src/Detail/Commanding/Command/Listing/Filter.php [hunk 7]
```diff
@@ -95,9 +107,20 @@ public function getValue()
...
+    public function setValue($value, $type = null)
     {
+        if ($type !== null) {
+            $this->setType($type);
+        }
+
+        // If type is set cast value to it
+        if ($this->getType() !== null) {
+            // No need to check that getMainType result is not null
+            settype($value, $this->getMainType($this->getType()));
```
Comment (ivan-wolf) [inline src/Detail/Commanding/Command/Listing/Filter.php]: `$this->setType()`?

**Pull request merged**